src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingSearchServiceTest.kt

```diff
@@ -320,28 +514,289 @@ class BookingSearchServiceTest {
+    every { mockUserService.getUserForRequest() } returns UserEntityFactory()
+      .withYieldedProbationRegion {
+        ProbationRegionEntityFactory()
+          .withYieldedApArea {
+            ApAreaEntityFactory().produce()
+          }
+          .produce()
+      }
+      .produce()
+    every { mockBookingRepository.findBookings(any(), any(), any(), any()) } returns PageImpl(
+      listOf(
+        TestBookingSearchResult(),
+        TestBookingSearchResult(),
+        TestBookingSearchResult(),
+      ),
+    )
+    every { mockOffenderService.getOffenderByCrn(any(), any()) } returns AuthorisableActionResult.Success(
+      OffenderDetailsSummaryFactory().produce(),
+    )
+
+    val (result, metaData) = bookingSearchService.findBookings(
+      ServiceName.temporaryAccommodation,
+      null,
+      sortOrder,
+      BookingSearchSortField.personName,
+      1,
+    )
 
     assertThat(result is AuthorisableActionResult.Success).isTrue
-    result as AuthorisableActionResult.Success
     assertThat(result.entity is ValidatableActionResult.Success).isTrue
     val validationResult = result.entity as ValidatableActionResult.Success
     assertThat(validationResult.entity).hasSize(3)
 
-    assertThat(validationResult.entity.map { it.bookingCreatedAt }).isSortedAccordingTo { a, b ->
+    assertThat(validationResult.entity.map { it.personName }).isSortedAccordingTo { a, b ->
       when (sortOrder) {
         SortOrder.ascending -> compareValues(a, b)
         SortOrder.descending -> compareValues(b, a)
       }
     }
+    assertThat(metaData).isNotNull()
+    verify(exactly = 1) {
+      mockBookingRepository.findBookings(any(), any(), any(), any())
+    }
+    verify(exactly = 3) {
+      mockOffenderService.getOffenderByCrn(any(), any())
+    }
+  }
+
+  @Test
+  fun `findBookings returns empty results from repository when page number is given`() {
+    every { mockUserService.getUserForRequest() } returns UserEntityFactory()
+      .withYieldedProbationRegion {
+        ProbationRegionEntityFactory()
+          .withYieldedApArea {
+            ApAreaEntityFactory().produce()
+          }
+          .produce()
+      }
+      .produce()
+    every { mockBookingRepository.findBookings(any(), any(), any(), any()) } returns PageImpl(emptyList())
+
+    val (result, metaData) = bookingSearchService.findBookings(
+      ServiceName.temporaryAccommodation,
+      null,
+      SortOrder.ascending,
+      BookingSearchSortField.bookingCreatedAt,
+      1,
+    )
+
+    assertThat(result is AuthorisableActionResult.Success).isTrue
+    assertThat(result.entity is ValidatableActionResult.Success).isTrue
+    val validationResult = result.entity as ValidatableActionResult.Success
+    assertThat(validationResult.entity).hasSize(0)
+    assertThat(validationResult.entity).allMatch {
+      it.personName != null
+    }
```

Comment:
I think this assertion doesn't do anything given that it's already been asserted to be empty on the line above.

Reply:
agree, done.